modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -854,11 +855,16 @@ def join(
             right_on
         ), "'left_on' and 'right_on' lengths don't match"
 
-        for col in left_on:
-            assert col in self.columns, f"'left_on' references unknown column {col}"
-        for col in right_on:
-            assert col in other.columns, f"'right_on' references unknown column {col}"
+        def validate(what, df, col_names):
+            df_cols = df.columns
+            for col in col_names:
+                if col not in df_cols:
+                    if col in df.index.names:
+                        raise NotImplementedError("Merge by index")
```

Comment:
What is wrong with a merge by index?

Reply:
Thanks, it was my misunderstanding. Should be fixed now.